test/benchgc/README.md

```diff
@@ -0,0 +1,103 @@
+* pybind11
+
+## Build and install
+```
+# Please execute at the top level of the project
+
+mkdir -p build
+cd build
+
+cmake .. -DMLIR_DIR=$MLIR_PATH -DGC_TEST_ENABLE=ON -DGC_ENABLE_BINDINGS_PYTHON=ON -DGC_BENCH_ENABLE=ON
+make -j benchgc
+
+python -m pip install test/benchgc/dist/benchgc-*.whl
+
+```
+
+## Synopsis
+```
+python -m benchgc [OPTIONS] --driver [DRIVER] --case [CASE]
+```
+## Flags
+###  --driver [str]
+* linalg: test the single op in linalg dialect
+* mlir: upload a mlir file and run
+* pattern: predefined pattern test such as mlp
+
+### --case [str]
+* if driver=mlir, please provide a mlir file here to test
+* if driver=pattern, please provide the pre-defined pattern name, such as mlp here
+* if driver is a dialect name, please provide the detail op name to start a single op test
+
+### --seed [int]
+* set the seed to generate the test data and reprodce the test
+
+### --verbose [int]
+* set the verbose level
+
+### --md index:SHAPExTYPE
+* Describe the shape and data type for argument
+* Not available when driver=mlir
+* index means the order of argument, including both inputs and outs
+* use prefix `0x` (e.g. `0xbf16`) to represent 0d memref or tensor input
+* use data type directly (e.g.`f32`) to represent a normal scalar
+
+```
+# %arg0             -> index = 0
+# tensor<2x2x2xf32> -> index = 1
+
+module {
+  func.func @entry(%arg0: f32) -> tensor<2x2x2xf32> attributes {llvm.emit_c_interface} {
+    %0 = tensor.empty() : tensor<2x2x2xf32>
+    %1 = linalg.fill ins(%arg0 : f32) outs(%0 : tensor<2x2x2xf32>) -> tensor<2x2x2xf32>
+    return %1 : tensor<2x2x2xf32>
+  }
+}
+```
+
+### --fill index:fill_type:[:fill_parameter]*
+* If not set, benchgc will assign a default method for the argument
+
+| description | fill_type | fill_parameter |
+|-------------|-----------|-----------|
+| Zero | Z | |
+| Normal | N | mean, std |
+| Poisson | P | lambda |
+| Binomial | B | n, p |
+| Uniform | U | a, b |
+| Integer | I | a, b |
+| Pytorch tensor dump | F | dump filename |
+| Benchdnn driver | D | driver_name[:driver filling parameter]* |
+
+#### Benchdnn driver filling
+
+| driver_name | driver filling parameter |
+|-------------|--------------------------|
+| binary | src0/src1:src0 dtype:src1 dtype:dst dtype |
+| conv | src/wei:src dtype:wei dtype:dst dtype:amplifier |
+| eltwise | algorithm: alpha: beta (please check https://oneapi-src.github.io/oneDNN/dev_guide_eltwise.html)  |
+| matmul | src/wei:src dtype:wei dtype:dst dtype:amplifier |
+| pool | not required |
+
+### --cmp index:cmp_type:[:cmp_parameter]*
+* If not set, benchgc will assign a default method for the argument
+
+| description | cmp_type | cmp_parameter |
+|-------------|-----------|-----------|
+| P2P check | P | threshold, zero_percent(mistrust check) |
+| Norm check | N | threshold |
+| Benchdnn driver | D | driver_name:dtype:case |
+
+## Example
+```
+python3 -m benchgc --verbose 0 --driver linalg --case add --md 0:4x5x6xf32 --md 1:4x5x6xf32 --md 2:4x5x6xf32
```

Comment:
could you add a more complex example with the use of `filling` and `cmp`? and could you also add an example of the result output here?

Reply:
@xurui1995 
I have updated the example in the new version. Please have another review. I am not sure whether it satisfies your requirement.